### DISCUSSION TITLE
`simplify(expr, rules, {}, { exactFraction: true })` not working for (only) custom rules.

### DISCUSSION THREAD

**Discussion by JacobBeaudry** [josdejong/mathjs#2611] [Q&A] [answered]
Hi Jos!

Currently something like:

```
simplify('x * 0.25', [ '0 + n -> n' ], {}, { exactFractions: true } )
```
returns `x * 0.25`, whereas: 

```
simplify('x * 0.25', {} , { exactFractions: true } )
```
or

```
simplify('x * 0.25', [...simplify.rules, '0 + n -> n'], {} , { exactFractions: true } )
```
returns `x * 1 / 4`.

That is, it seems as though the `exactFractions` option does not work if one only invokes `simplify` with an array of _custom_ rules, which suggests that the conversion to exact fractions is (may be) currently implemented using the built-in simplifications rules.

 **Is it possible to get exact fractions when only applying custom simplification rules?**

Thanks in advance! Hope all's well on your end!

Jacob

**Comment by josdejong** [comment, marked as answer]
The `exactFractions` option is an option of one of the built-in rules (a function rule `simplifyConstant`). If you clear all rules, the rule handling `exactFractions` is not executed. This rule currently not exposed and only used internally in `simplify`.

As a workaround, you can find the `simplifyConstant` from `simplify.rules` since it is (currently) the only function rule:

```js
const simplifyConstant = math.simplify.rules.find(rule => typeof rule === 'function')
```
However this is not guaranteed to work in the future.

So I think we have to do the following

1. Improve the docs of `exactFractions` to explain that when you clear out all rules, that this option is not executed
2. Expose the rule somehow so you can use it on your own. Expose the factory function `createSimplifyConstant`, or maybe expose a new property `math.simplify.simplifyConstant` or so.
3. We _could_ implement warnings or so when using `exactFractions` whilst `simplifyConstant` is missing from the rules, but that feels a bit far-fetched to me.


Yeah all going fine here, thanks for asking :)
